lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -459,19 +473,24 @@ mkUnsignedTx ttl cs withdrawals certs =
         unsigned
 
 mkWithdrawals
-    :: ChimericAccount
+    :: forall (n :: NetworkDiscriminant). (Typeable n)
+    => Proxy n
+    -> ChimericAccount
     -> Word64
     -> Map (SL.RewardAcnt TPraosStandardCrypto) SL.Coin
-mkWithdrawals (ChimericAccount keyHash) amount
+mkWithdrawals _ (ChimericAccount keyHash) amount
     | amount == 0 = mempty
     | otherwise = Map.fromList
-        [ ( SL.RewardAcnt SL.Mainnet keyHashObj
+        [ ( SL.RewardAcnt network keyHashObj
           , SL.Coin $ fromIntegral amount
           )
         ]
   where
     keyHashObj = SL.KeyHashObj $ SL.KeyHash $ Hash.UnsafeHash keyHash
-
+    network =
+        case testEquality (typeRep @n) (typeRep @'Mainnet) of
+            Just{}  -> SL.Mainnet
+            Nothing -> SL.Testnet
```

Comment:
relevant part is here :point_up:

Reply:
just wondering if this part should not go to Compatibility, we have there number of network translation like  https://github.com/input-output-hk/cardano-wallet/blob/master/lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs#L882 or https://github.com/input-output-hk/cardano-wallet/blob/master/lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs#L572

Reply:
Ho yes, it probably should. Happy to do it on Monday :sweat_smile: